Android 6 permissions

Semrubo opened the pull request:
VATRP-1565, VATRP-1574, VATRP-1575, VATRP-1576

Commit:
Merge remote-tracking branch 'refs/remotes/VTCSecureLLC/master'

Commit:
Merge remote-tracking branch 'refs/remotes/VTCSecureLLC/master' into android_6_permissions

Commit:
Crash fix for marshmallow permissions

Besides crash fixes related to new permissions module in Android
Marshmallow.
Gradle is changed to be compiled with api 23.
v4 support jar  is removed from libs.
Appcompat-v7 dependency is added.
Notification for Vcards is changed to work on all android versions.
Checks for camera permission on dialer screen.

Commit:
Instant messaging crash fix for marshmallow

Don't attack file to instant messing when access to storage is not
granted

Commit:
Crash fix related to Camera

Check permission to the camera when getting camera config.

Commit:
Ask for contacts permission when contacts screen is opened

If there is no permission granted for contacts and contacts screen is
opened ask for permission.

Commit:
Check permission for outgoing call

If Mic permission is not granted and outgoing call is being initialized,
cancel it and ask for Mic permission

Commit:
fix to the issue "no permission to mic when receiving a call"

When a call is received and there is no permission for mic, show the
informing dialogue, and decline the call, as soon as OK button is
pressed.

Commit:
Contact crash fix for android 6

Commit:
Contact crash for android 5

Commit:
Crash fix related to LinphoneActivity

When application is restarted for permission changes, there were cases
when application uses not initialized object of LinphoneActivity

shareefalis commented:
Do a revert of this commit since you got the new permission  model working fine. https://github.com/shareefalis/ace-android/commit/ec3ab3b74667f896d7bf7cf6c1c9b2aaf1defea9

Commit:
Merge remote-tracking branch 'refs/remotes/VTCSecureLLC/master' into android_6_permissions

# Conflicts:
#	src/org/linphone/DialerFragment.java

Commit:
Merge remote-tracking branch 'refs/remotes/VTCSecureLLC/master' into android_6_permissions

Commit:
Makefile and manifest file changes for api target 23

BUILD_AMRWB is disabled as it has text relocation issue.
And target api changed from 22 to 23 in manifest to solve Travis build
error.

Commit:
Merge remote-tracking branch 'refs/remotes/VTCSecureLLC/master' into android_6_permissions

Commit:
gitignore changes to exclude support libs from git

android-support-v4.jar and android-support-v7-appcompat.jar are removed
from gitignore to be added in repo

Commit:
Config changes

Gradle is changed to compile support libs.
Added support files libs.

Commit:
Support libs

support Libs added

Commit:
Libs added to gitignore

Libs folder added to gitignore

kennethdick commented:
+1
